quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ManagementPropertyMappers.java

```diff
@@ -117,7 +120,13 @@ public static PropertyMapper<?>[] getManagementPropertyMappers() {
     }
 
     public static boolean isManagementEnabled() {
-        return isTrue("quarkus.management.enabled");
+        if (isNonServerMode()) return false;
+        var isManagementOccupied = isTrue(HealthOptions.HEALTH_ENABLED) || isTrue(MetricsOptions.METRICS_ENABLED);
+        return isManagementOccupied && !isTrue(ManagementOptions.LEGACY_OBSERVABILITY_INTERFACE);
```

Comment:
```suggestion
        if (isNonServerMode() || isTrue(ManagementOptions.LEGACY_OBSERVABILITY_INTERFACE)) {
            return false;
        }
        return isTrue(HealthOptions.HEALTH_ENABLED) || isTrue(MetricsOptions.METRICS_ENABLED);
```

Might be simplier to put the false conditions together.

Reply:
Thanks, I've updated it. But I left the named var there to emphasize when the mgmt interface is enabled. @shawkins Is it ok?